cmd/kubeadm/app/componentconfigs/config.go

```diff
@@ -55,14 +55,33 @@ func GetFromKubeletConfigMap(client clientset.Interface, version *version.Versio
 	return obj, nil
 }
 
+// Custom error for handling kube-proxy ConfigMap errors
+type kubeProxyConfigMapError struct {
+	s string
+}
+
+func (e *kubeProxyConfigMapError) Error() string {
+	return e.s
+}
+
+func newKubeProxyConfigMapError(err error) *kubeProxyConfigMapError {
+	return &kubeProxyConfigMapError{err.Error()}
+}
+
+// IsKubeProxyConfigMapError returns true if the given error is of type kubeProxyConfigMapError
+func IsKubeProxyConfigMapError(err error) bool {
+	_, ok := err.(*kubeProxyConfigMapError)
+	return ok
+}
+
 // GetFromKubeProxyConfigMap returns the pointer to the ComponentConfig API object read from the kube-proxy
 // ConfigMap map stored in the cluster
 func GetFromKubeProxyConfigMap(client clientset.Interface, version *version.Version) (runtime.Object, error) {
 
 	// Read the ConfigMap from the cluster
 	kubeproxyCfg, err := client.CoreV1().ConfigMaps(metav1.NamespaceSystem).Get(kubeadmconstants.KubeProxyConfigMap, metav1.GetOptions{})
 	if err != nil {
-		return nil, err
+		return nil, newKubeProxyConfigMapError(err)
```

Comment:
Do we need to detect error type here?
Like:
```
apierrors.IsNotFound(err)
```

Reply:
it can work.
if `IsNotFound` is used we can technically remove `kubeProxyConfigMapError`